### ISSUE TITLE
Deletion modal doesn't use the UI kit

### ISSUE THREAD

**Issue by MatShir** [PrestaShop/PrestaShop#26936]
**Describe the bug and add screenshots**
Modal deletion of cart rule page doesn't use the UI kIt

<img width="1220" alt="Capture d’écran 2021-12-09 à 15 22 36" src="https://user-images.githubusercontent.com/43613217/145416700-e5c6b073-2791-4750-b7e7-1ce6d9bdc80e.png">

**Expected behavior**
Use the UI kit

**Steps to reproduce**

1. **BO**>**SELL**>**Orders**>**Discount**
2. Click on **Add new cart rule** button
3. Fill all infos
4. Save
5. Back to the list
6. Click on **Delete** button
7. See error > the modal doesn't use UI kit

_Originally posted by @HanaRebaiQA in https://github.com/PrestaShop/PrestaShop/issues/26514#issuecomment-983715756_

**Comment by mparvazi** [comment]
This page (Cart Rules) designed in legacy theme.
I think the page must be migrated to new theme to solve this issue.
The grid tables that has action buttons in 3 dots style, has been designed by new theme.

![Screenshot 2021-12-15 at 13-41-27 Pages • preload](https://user-images.githubusercontent.com/85633460/146167275-46098e82-c6d7-4e61-9c75-1a18134ff097.png)

**Comment by mparvazi** [comment]
The pages designed in legacy theme:

Order Settings > Statuses 
Customer Settings >  Groups 
Customer Settings > Titles 
Contact > Stores 
Search > Tags 
Locations > Countries 
Locations > States 
Taxes > Tax Rules 
Shipping > Carriers 
Theme & Logo > Pages Configuration 
Design > Image Settings 
Customer Service > Customer Service 
Customer Service > Merchandise Returns 
Attributes & Features > Attributes 
Attributes & Features > Features 
Discounts > Cart Rules 
Discounts > Catalog Price Rules 
Orders > Shopping Carts
Dashboard

**Comment by riyasavant** [comment]
Can I work on this issue?

**Comment by MatShir** [comment]
Of course, you can work on the issue. We are happy to welcome new motivated contributor 🤗 
Keep in mind that some of the page have been migrated, so the list might not be up-to-date. If you are willing to make a PR, I could double-check to update the list. Let me know :)

**Comment by MatShir** [comment]
You can target 8.1.x branch !

**Comment by boo-code** [comment]
Still reproduces on `9.2.x`, and it is narrower than the thread suggests - it does not need the page migrated.

The Discounts page is indeed still legacy: `src/PrestaShopBundle/Controller/Admin/Sell/Catalog/CartRuleController.php` exists but carries only `searchAction()`, the autocomplete endpoint for order creation, so the list is still served by `controllers/admin/AdminCartRulesController.php`. (`ps_tab.route_name` is empty for this tab, but that is true of most migrated pages too, so it is not a migration signal either way.)

The modal itself is not per-page. `admin-dev/themes/default/template/helpers/list/list_action_delete.tpl` calls `confirm_link(...)`, which is defined once, at `js/admin.js:1323`, and delegates to `jConfirm` from `js/jquery/plugins/alerts/jquery.alerts.js` - jQuery Alert Dialogs 1.1, dated 14 May 2009.

`jConfirm` has exactly one call site in the whole tree, that one. So replacing it inside `confirm_link()` with a UI-kit modal fixes the deletion modal on every legacy list at once, rather than on Cart Rules alone - which also answers the list of pages in the comment above.

Worth keeping separate: `jAlert` from the same plugin is still called from about a dozen places (`js/admin/import.js`, `carrier_wizard.js`, `products.js`, `dashboard.js`, the groups and customer-threads templates). Those are error alerts rather than the deletion modal, so they are a different change.

Leaving this one alone, since `Good first issue` and `Hacktoberfest` are on it and the single-call-site shape above should make it approachable for whoever picks it up.
